app/controllers/searches_controller.rb

```diff
@@ -6,8 +6,9 @@ def show
     druid = params.require(:q)
     druid = "druid:#{druid}" unless druid.start_with?('druid:')
     result = Work.find_by(druid: druid) || Collection.find_by(druid: druid)
-    return render js: "window.location='#{polymorphic_path(result)}'" if result
+    return render json: { message: 'not found' }, status: :not_found unless result
 
-    render json: { message: 'not found' }, status: :not_found
+    path = result.instance_of?(Work) ? work_path(result) : collection_version_path(result.head)
+    render js: "window.location='#{path}'"
```

Comment:
Had to remove polymorphic_path since we're changing which tab to go to for a collection.